dist/init/linux-upstart/caddy.conf

```diff
@@ -1,18 +1,25 @@
-description "Caddy startup script"
+description "Caddy HTTP/2 web server"
 author "Mathias Beke"
 
 start on runlevel [2345]
 stop on runlevel [016]
 
+console log
 
 setuid www-data
 setgid www-data
 
 respawn
 respawn limit 10 5
 
-limit nofile 4096 4096
+# Letsencrypt-issued certificates will be written to this directory.
+env HOME=/etc/ssl/caddy
+
+limit nofile 1048576 1048576
 
 script
-	exec /usr/bin/caddy -agree=true -conf=/etc/caddy/Caddyfile
-end script
\ No newline at end of file
+        mkdir -p /etc/caddy
```

Comment:
In general, at least in the "old days", startup scripts and such should not  modify files in `/etc/...`.  In some places, these filesystems are read-only.  Better options were:

- `/var/run/...`
- `/tmp`
- `/var/tmp`

Each with their own semantic differences (such as `/tmp` being wiped out at times, etc).